src/argilla/client/feedback/dataset/base.py

```diff
@@ -306,7 +311,12 @@ def unify_responses(
     def prepare_for_training(
         self,
         framework: Union[Framework, str],
-        task_mapping: TrainingTaskMappingForTextClassification,
+        task: Union[
+            TrainingTaskForTextClassification,
+            TrainingTaskForSupervisedFinetuning,
+            TrainingTaskForRewardModelling,
+            TrainingTaskForDirectPreferenceOptimization,
+        ],
         train_size: Optional[float] = 1,
```

Comment:
I thought the standard was to define the `Optional` on fields that are `None` by default, as Python will mark those as optional internally if those have a predefined value as `float = 1`, so I would change this to

```suggestion
        train_size: float = 1.0,
```

Reply:
`Optional[X]` just means `Union[None, X]`. So, this says `Union[None, float]`. When looking at the code, I think that's true. E.g. the `_prepare_for_training_with_transformers` accepts None as a valid parameter.